File: leetcode/js/remove-trailing-zeros-from-a-string.js
/**
 * @param {string} num
 * @return {string}
 */
var removeTrailingZeros = function (num) {
  const numArr = [...num];

  while (numArr.at(-1) === "0") {
    numArr.pop();
  }

  return numArr.join("");
};

var removeTrailingZeros = function (num) {
  return num.replace(/0+$/, "");
};

File: leetcode/rust/remove-trailing-zeros-from-a-string.rs
impl Solution1 {
    pub fn remove_trailing_zeros(num: String) -> String {
        let mut num_arr: Vec<char> = num.chars().collect();

        while num_arr[num_arr.len() - 1] == '0' {
            num_arr.pop();
        }

        num_arr.iter().collect()
    }
}

impl Solution2 {
    pub fn remove_trailing_zeros(num: String) -> String {
        let mut i = num.len() - 1;

        while num.chars().nth(i) == Some('0') {
            i -= 1;
        }

        num[..i + 1].to_string()
    }
}
